src/bitboard.rs

```diff
@@ -52,20 +52,20 @@ pub struct Bitboard(pub u64);
 impl Bitboard {
     /// A bitboard with a single square.
     #[inline]
-    pub fn from_square(sq: Square) -> Bitboard {
-        Bitboard(SQUARES[usize::from(sq)])
+    pub const fn from_square(sq: Square) -> Bitboard {
+        Bitboard(SQUARES[sq as usize])
```

Comment:
afaik this is equivalent to `usize::from(sq)` but one worked and the other didn't

Reply:
`as` can silently become lossy when refactoring, so generally `From::from` is preferable for conversions. But using traits in const context would require [`#![feature(const_trait_impl)]`](https://doc.rust-lang.org/beta/unstable-book/language-features/const-trait-impl.html).